community/kernel/src/main/java/org/neo4j/unsafe/batchinsert/internal/BatchInserterImpl.java

```diff
@@ -234,6 +224,7 @@ public BatchInserterImpl( final File storeDir, final FileSystemAbstraction fileS
 
         life = new LifeSupport();
         this.storeDir = storeDir;
+        life.add( new FileSystemLifecycleAdapter( fileSystem ) );
```

Comment:
The pattern so far has been "if you created the FSA you are responsible for closing it". This breaks that pattern.
Example:
`BatchInserterTest` uses a `DefaultFileSystemRule` and provides a FSA from this rule when creating a `BatchInserterImpl` using `BatchInserters`.
Now, they will both try to close the FSA which is fine but it is problematic that you can not trust your FSA not to be closed by someone else when giving it away.

I suggest `BatchInserters` should always take FSA from outside and never create its own, and thus we can remove this line.